centreon/www/class/centreonGraph.class.php

```diff
@@ -211,35 +211,45 @@ public function __construct($user_id, $index = null, $debug = 0, $compress = nul
         unset($opt);
 
         /* Get RRDCacheD options */
-        $result = $this->DB->query(
+        $statement = $this->DB->prepare(
             "SELECT config_key, config_value
-            FROM cfg_centreonbroker_info AS cbi
-            INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
-            INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
-            WHERE ns.localhost = '1'
-            AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
+             FROM cfg_centreonbroker_info AS cbi
+             INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
+             INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
+             WHERE ns.localhost = '1'
+             AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
         );
-        while ($row = $result->fetch()) {
+        $statement->execute();
+        $result = $statement->fetchAll(PDO::FETCH_ASSOC);
+
+        foreach ($result as $row) {
             $this->rrdCachedOptions[$row['config_key']] = $row['config_value'];
         }
+        $statement->closeCursor();
 
         if (isset($index)) {
-            $DBRESULT = $this->DB->query("SELECT `metric_id`
-                                          FROM `ods_view_details`
-                                          WHERE `index_id` = '" . $this->index . "'
-                                          AND `contact_id` = '" . $this->user_id . "'");
+            $DBRESULT = $this->DB->prepare("SELECT `metric_id`
+              FROM `ods_view_details`
+              WHERE `index_id` = :index_id
+              AND `contact_id` = :user_id");
+            $DBRESULT->bindParam(':index_id', $this->index, PDO::PARAM_INT);
+            $DBRESULT->bindParam(':user_id', $this->user_id, PDO::PARAM_INT);
```

Comment:
There's no need to link PHP variables by reference here.
```suggestion
            $DBRESULT->bindValue(':index_id', $this->index, PDO::PARAM_INT);
            $DBRESULT->bindValue(':user_id', $this->user_id, PDO::PARAM_INT);
```